aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -98,6 +101,75 @@ Tensor _dim_arange(const Tensor& like, int64_t dim) {
   return at::arange(like.size(dim), like.options().dtype(at::kLong));
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~ complex / complex_polar ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+void complex_check_floating(const Tensor& a, const Tensor& b) {
+  TORCH_CHECK((a.scalar_type() == kFloat || a.scalar_type() == kDouble) &&
+              (b.scalar_type() == kFloat || b.scalar_type() == kDouble),
+              "Both inputs must be either Float or Double");
```

Comment:
"Expected both inputs to be float or double tensors, but got ..."

Similar check for reference: 

https://github.com/pytorch/pytorch/blob/ba98c0e38ca76a8e03d27e20a1d2bfc324d502ca/aten/src/ATen/native/cuda/Lerp.cu#L11